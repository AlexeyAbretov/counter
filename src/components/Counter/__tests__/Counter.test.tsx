/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { it, describe, expect } from 'vitest';

import { theme } from '@theme';
import { Counter, CounterType } from '..';

const getUi = (title: string, type: CounterType): JSX.Element => {
  return (
    <ThemeProvider theme={theme}>
      <Counter title={title} type={type} />
    </ThemeProvider>
  );
};

describe('Counter tests', () => {
  it('should render zero type', () => {
    const { container } = render(getUi('0', 'zero'));

    expect(container).toMatchSnapshot();
  });

  it('should render lesszero type', () => {
    const { container } = render(getUi('-1', 'lesszero'));

    expect(container).toMatchSnapshot();
  });

  it('should render abovezero type', () => {
    const { container } = render(getUi('1', 'abovezero'));

    expect(container).toMatchSnapshot();
  });
});
