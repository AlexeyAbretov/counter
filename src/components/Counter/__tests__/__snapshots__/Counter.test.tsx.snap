// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter tests > should render abovezero type 1`] = `
<div>
  <div
    class="sc-bcXHqe hHWHmU"
    data-testid="counter"
    type="abovezero"
  >
    1
  </div>
</div>
`;

exports[`Counter tests > should render lesszero type 1`] = `
<div>
  <div
    class="sc-bcXHqe jXLxPF"
    data-testid="counter"
    type="lesszero"
  >
    -1
  </div>
</div>
`;

exports[`Counter tests > should render zero type 1`] = `
<div>
  <div
    class="sc-bcXHqe ftiUJn"
    data-testid="counter"
    type="zero"
  >
    0
  </div>
</div>
`;
